torch/csrc/autograd/functions/convolution.cpp

```diff
@@ -342,6 +379,136 @@ auto ConvBackward::releaseVariables() -> void {
   bias_.data.reset();
 }
 
+
+// ConvBackwardBackward implementation
+
+auto ConvBackwardBackward::apply(const variable_list& grad_grad_inputs) -> variable_list {
+  check_input_variables("ConvNdBackwardBackward", grad_grad_inputs, 3, 0);
+  if (transposed) throw std::runtime_error("ConvBackwardBackward does not support transposed convolution");
+
+  auto ggI = grad_grad_inputs[0];
+  auto ggW = grad_grad_inputs[1];
+  auto ggb = grad_grad_inputs[2];
+
+  auto gO = grad_output_.unpack();
+  auto weight = weight_.unpack();
+
+  // Compute ggO = conv(w, ggI) + conv(ggW, i) + ggb
+  std::shared_ptr<Variable> ggO = nullptr;
+  if (ggI) {
+    if (weight->data->isCuda()) {
+      weight = std::make_shared<Contiguous>()->apply({weight})[0];
+    }
+    ggO = std::make_shared<ConvForward>(*this)->apply({ggI, weight, nullptr})[0];
+  }
+
+  if (ggW) {
+    if (ggW->data->isCuda()) {
+      ggW = std::make_shared<Contiguous>()->apply({ggW})[0];
+    }
+    auto ggW_term = std::make_shared<ConvForward>(*this)->apply({input_.unpack(), ggW, nullptr})[0];
+    if (ggO) {
+      ggO = std::make_shared<Add>()->apply({ggO, ggW_term})[0];
+    } else {
+      ggO = ggW_term;
+    }
+  }
+
+  if (ggb) {
+    // View as (1, ggb.size(0), 1, 1...)
+    std::vector<long> new_size(gO->data->nDim(), 1);
+    new_size[1] = ggb->data->sizes()[0];
+    auto ggb_contiguous = std::make_shared<Contiguous>()->apply({ggb})[0];
+    auto ggb_view = std::make_shared<View>(new_size)->apply({ggb_contiguous})[0];
+
+    // Expand 
+    auto ggb_expanded = std::make_shared<Expand>(gO->data->sizes())->apply({ggb_view})[0];
+
+    if (ggO) {
+      ggO = std::make_shared<Add>()->apply({ggO, ggb_expanded})[0];
+    } else {
+      ggO = ggb_expanded;
+    }
+  }
+
+  // Compute gW = conv(gO, ggI)
+  std::shared_ptr<Variable> gW = nullptr;
+  if (ggI) {
+    // Modified params with correct padding
+    ConvParams gw_conv_params(*this);
+    auto weight_size = weight->data->sizes();
+    std::vector<long> kernel_size(weight_size.begin() + 2, weight_size.end());
+    auto input_size = ggI->data->sizes();
+    std::vector<long> input_shape(input_size.begin() + 2, input_size.end());
+    for(size_t i=0; i<gw_conv_params.padding.size(); ++i) {
+      // Formula for conv output size before the floor operation
+      auto out_size = float(input_shape[i] + 2 * gw_conv_params.padding[i] -
+                      gw_conv_params.dilation[i] * (kernel_size[i] - 1) - 1) /
+                      gw_conv_params.stride[i] + 1;
+      auto exact_out_size = floorf(out_size);
```

Comment:
I need to check if the division is exact or not and then use the integer part.
So I use the float version and the integer version to check if the division is exact and then use the integer version.
Do you think of a better way to do this?